sympy/core/relational.py

```diff
@@ -1075,9 +1079,281 @@ def _n2(a, b):
+    if l.is_Pow and r.is_Pow:
+        if l == r:
+            if m == mr:
+                L, R = b - br, S.Zero
+                if ix:
+                    L, R = R, L
+                return L >= R
+            L = l
+            R = br - b
+            den = m - mr
+            if den < 0:
+                ix = ~ix
+            R /= den
+        elif b == br:
+            if l.base == r.base:
+                l = Pow(l.base, l.exp - r.exp, evaluate=False)
+                r = mr/m
+                if is_ge(r, S.Zero) and m > 0:
+                    L, R = l, r
+                elif is_ge(r, S.Zero) and m < 0:
+                    L, R = l, r
+                    ix = ~ix
+                elif is_ge(S.Zero, r) and m > 0:
+                    L, R = l, r
+                    ix = ~ix
+                elif is_ge(S.Zero, r) and m < 0:
+                    L, R = l, r
+                if L != l or R != r:
+                    return
+            elif l.exp == r.exp and l.exp.is_integer or l.base.is_nonnegative and r.base.is_positive:
+                L = (l.base/r.base)**l.exp
+                R = mr/m
+                if m < 0:
+                    ix = ~ix
+    elif l.is_Pow:
+        L = l
+        R = mr/m*r + (br - b)/m
+        if m < 0:
+            ix = ~ix
+    elif r.is_Pow:
+        L = r
+        R = m/mr*l + (b - br)/mr
+        ix = ~ix
+        if mr < 0:
+            ix = ~ix
+    else:
+        L = l
+        R = mr/m*r + (br - b)/m
+        if m < 0:
+            ix = ~ix
+
+    if R.is_Number:
+        n, d = fraction(L)
+        if d is not S.One:
+            # bring denom to numerator on other side
+            # n/d ~ R -> n/R ~ d, watching for signs of R and d
+            if R != 0:
+                if R < 0:
+                    ix = ~ix
+                den = d
+                r = R
+            else:
+                den = S.Zero
+                r = S.One
+            if is_ge(d, S.Zero):
+                L, R = n/r, den
+            elif is_le(d, S.Zero):
+                ix = ~ix
+                L, R = n/r, den
+
+    if ix:  # the sense of the comparison is reversed
+        L, R = R, L
+
+    # if we did more than change sign or switch sides, start again
+    if {L, R} - {lhs, rhs, -lhs, -rhs}:
+        return is_ge(L, R)
+
+    # check for results based on regions
+    if not any(i in (L, R) for i in (-1, 0, 1)):
+        if is_ge(L, S.One):
+            if is_ge(S.NegativeOne, R):
+                return True
+            if is_ge(S.Zero, R):
+                return True
+            if is_ge(S.One, R):
+                return True
+        if is_ge(L, S.Zero):
+            if is_ge(S.NegativeOne, R):
+                return True
+            if is_ge(S.Zero, R):
+                return True
+        if is_ge(L, S.NegativeOne):
+            if is_ge(S.NegativeOne, R):
+                return True
+
+    if L.is_Pow or R.is_Pow:
+        b, e = L.as_base_exp()
+        B, E = R.as_base_exp()
```

Comment:
The idea with multipledispatch is supposed to be that we can write a `Pow/Pow` handler rather than having all the logic in an `Expr/Expr` handler that checks the types.